netutils/forward.go

```diff
@@ -23,102 +22,66 @@ type Conn interface {
 
 func ForwardConnection(ctx context.Context, srcSocket net.Conn, ns, ip, port string) error {
 	log := logger.Get(ctx)
-	addr, err := getTempFilename()
+
+	runtime.LockOSThread()
+	defer runtime.UnlockOSThread()
+
+	dst, err := netns.GetFromPath(ns)
 	if err != nil {
-		return errors.Wrapf(err, "fail to get temp file name for unix socket")
+		return errors.Wrapf(err, "fail to get dest namespace handler %v", ns)
 	}
+	defer dst.Close()
 
-	src := srcSocket.(Conn)
+	current, err := netns.Get()
+	if err != nil {
+		return errors.Wrapf(err, "fail to get current namespace handler")
+	}
+	defer current.Close()
 
-	// Explanation of the following tricky part
-	// At first we wanted to forward the socket from the HTTP connection directly
-	// to the child process located in the namespace of the sand network, thus
-	// able to reach the final target.
-	//
-	// This method doesn't work as when sand is running with TLS, the connection
-	// is a *tls.Conn and there is no way to transfer such socket in a child
-	// process.
+	err = netns.Set(dst)
+	if err != nil {
+		return errors.Wrapf(err, "fail to set current namespace to dst %v", dst)
+	}
 
-	// So we need to pass through an intermediary IPC system, here a UNIX socket.
-	// Basically the parent process is creating a unix socket server, the child
-	// process in the NS of our target is creating a connection to this socket,
-	// this connection is linking the remote client connection and the internal
-	// sand connection.
-	//
-	// HTTP client <----- inter-process unix conn -----> connection to target in sand network
-	unixSocket, err := net.ListenUnix("unix", &net.UnixAddr{Net: "unix", Name: addr})
+	dstHost := fmt.Sprintf("%s:%s", ip, port)
+	dstSocket, err := net.Dial("tcp", dstHost)
 	if err != nil {
-		return errors.Wrapf(err, "fail to open unix socket")
+		return errors.Wrapf(err, "fail to open connection to %v", dstHost)
 	}
-	defer unixSocket.Close()
 
+	wg := &sync.WaitGroup{}
+	wg.Add(2)
+
+	// Error logs are Info as we're only proxying connections, if one is stopped abruptely
+	// The real error is either from the client side or the destination side, it's not an
+	// error for the proxy itself
 	go func() {
-		clientSocket, err := unixSocket.AcceptUnix()
-		if err != nil {
-			log.WithError(err).Error("fail to accept unix connection")
+		defer wg.Done()
+		defer dstSocket.Close()
+		_, err := io.Copy(dstSocket, srcSocket)
+		if err != io.EOF {
+			log.Info("end of connection from unix src to dst socket with error: %v", err)
```

Comment:
You could write `log.WithError(err).Info("...")` so that the error message is in a key `error=` which makes the log line easier to parse with Kibana.

Anyway you should at least update the line to use `Infof` instead of `Info`